From: Shyam Prasad N <sprasad@microsoft.com>
Date: Sun, 16 Jan 2022 13:28:34 +0000
Subject: [PATCH] cifs: remove repeated state change in dfs tree connect
Git-commit: ece0767641740c7eea7aee5a332728e115b00eab
References: bsc#1193629
Patch-mainline: v5.17-rc1

cifs_tree_connect checks and sets the tidStatus for the tcon.
cifs_tree_connect also calls a dfs specific tree connect
function, which also does similar checks. This should
not happen. Removing it with this change.

Signed-off-by: Shyam Prasad N <sprasad@microsoft.com>
Signed-off-by: Steve French <stfrench@microsoft.com>
Acked-by: Enzo Matsumiya <ematsumiya@suse.de>
---
 fs/cifs/connect.c | 10 ----------
 1 file changed, 10 deletions(-)

diff --git a/fs/cifs/connect.c b/fs/cifs/connect.c
index 4a31a9b3f34f..6740a7c39df3 100644
--- a/fs/cifs/connect.c
+++ b/fs/cifs/connect.c
@@ -4295,16 +4295,6 @@ static int __tree_connect_dfs_target(const unsigned int xid, struct cifs_tcon *t
 	struct dfs_cache_tgt_iterator *tit;
 	bool target_match;
 
-	/* only send once per connect */
-	spin_lock(&cifs_tcp_ses_lock);
-	if (tcon->tidStatus != CifsNew &&
-	    tcon->tidStatus != CifsNeedTcon) {
-		spin_unlock(&cifs_tcp_ses_lock);
-		return 0;
-	}
-	tcon->tidStatus = CifsInTcon;
-	spin_unlock(&cifs_tcp_ses_lock);
-
 	extract_unc_hostname(server->hostname, &tcp_host, &tcp_host_len);
 
 	tit = dfs_cache_get_tgt_iterator(tl);
-- 
2.36.1


